iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java

```diff
@@ -771,16 +785,20 @@ public AudienceMembershipChangeResponse processAudienceMembershipChangeRequest(A
         return new AudienceMembershipChangeResponse();
     }
 
-    private void addUserIdentitiesToRequest(UserRequest request, List<UserIdentity> identities) {
+    private void addUserIdentitiesToRequest(UserRequest request, EventProcessingRequest processingRequest) {
+        List<UserIdentity> identities = processingRequest.getUserIdentities();
         if (identities != null) {
             for (UserIdentity identity : identities) {
                 if (identity.getType().equals(UserIdentity.Type.EMAIL)) {
                     request.email = identity.getValue();
-                } else if (identity.getType().equals(UserIdentity.Type.CUSTOMER)) {
+                } else if (identity.getType().equals(UserIdentity.Type.CUSTOMER) && !shouldUseMPID(processingRequest)) {
```

Comment:
I'd recommend converting this method to something that maps the value of the setting onto various identities, such as the other identities. right now this only supports customer id and mpid

Reply:
Are there best practices on how to handle this? Should we use something like `userIdField.equals(userIdentity.getType().toString())`?